turbopack/crates/turbopack-ecmascript/src/global_module_id_strategy.rs

```diff
@@ -164,58 +164,46 @@ pub async fn children_modules_idents(
     Ok(PreprocessedChildrenIdents { modules_idents }.cell())
 }
 
+const JS_MAX_SAFE_INTEGER: u64 = (1u64 << 53) - 1;
+
 // Note(LichuAcu): This could be split into two functions: one that merges the preprocessed module
 // ids and another that generates the final, optimized module ids. Thoughts?
 pub async fn merge_preprocessed_module_ids(
     preprocessed_module_ids: Vec<Vc<PreprocessedChildrenIdents>>,
 ) -> Result<HashMap<RcStr, Vc<ModuleId>>> {
-    let mut module_id_map: HashMap<RcStr, Vc<ModuleId>> = HashMap::new();
-    let mut used_ids: HashSet<u64> = HashSet::new();
+    let mut merged_module_ids = HashMap::new();
```

Comment:
The order of merged_module_ids need to be deterministic for the hashes to be deterministic.

Either use an `IndexMap` here or sort the `merged_module_ids` after merging